File: fastapi-server/src/modules/sample/application/queries/get_sample.py
"""sample モジュールのサンプル取得クエリ。"""

from src.modules.sample.domain import SampleNotFoundError, SampleRepository

from ..dto import GetSampleResponse


class GetSampleQuery:
    """サンプル取得クエリ

    リポジトリを使用してIDでサンプルエンティティを取得します。
    """

    def __init__(self, repository: SampleRepository) -> None:
        """リポジトリでクエリを初期化

        Args:
            repository: データアクセス用のサンプルリポジトリ
        """
        self._repository = repository

    async def execute(self, id: str) -> GetSampleResponse:
        """サンプル取得クエリを実行

        Args:
            id: 取得するサンプルID

        Returns:
            サンプルレスポンス DTO

        Raises:
            SampleNotFoundError: サンプルが見つからない場合
        """
        entity = await self._repository.find_by_id(id)

        if entity is None:
            raise SampleNotFoundError(id)

        return GetSampleResponse(
            id=entity.id,
            name=str(entity.name),
            description=entity.description,
            created_at=entity.created_at,
            updated_at=entity.updated_at,
        )

File: go-server/internal/modules/sample/application/queries/get_sample.go
package queries

import (
	"context"

	"cariban/go-server/internal/modules/sample/application"
	"cariban/go-server/internal/modules/sample/domain"
)

type GetSampleQuery struct {
	repository domain.SampleRepository
}

func NewGetSampleQuery(repository domain.SampleRepository) *GetSampleQuery {
	return &GetSampleQuery{repository: repository}
}

func (q *GetSampleQuery) Execute(ctx context.Context, id string) (application.GetSampleResponse, error) {
	entity, err := q.repository.FindByID(ctx, id)
	if err != nil {
		return application.GetSampleResponse{}, err
	}

	return application.GetSampleResponse{
		ID:          entity.ID,
		Name:        entity.Name.String(),
		Description: entity.Description,
		CreatedAt:   entity.CreatedAt,
		UpdatedAt:   entity.UpdatedAt,
	}, nil
}
